conans/model/graph_lock.py

```diff
@@ -158,19 +158,14 @@ def update_lock(self, new_lock):
         and raise
         """
         for id_, node in new_lock._nodes.items():
-            if node.modified:
+            if node.modified == BINARY_BUILD:
```

Comment:
Update the docstring of this method, it is not accurate anymore. Explain the "BINARY_BUILD" that after the merge is changed by "True"